src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -328,9 +328,11 @@ private static boolean isBlockAloneOnSingleLine(Details details) {
             nextToken = Details.getNextToken(nextToken);
         }
 
-        if (nextToken != null && nextToken.getType() == TokenTypes.DO_WHILE) {
-            final DetailAST doWhileSemi = nextToken.getParent();
-            nextToken = Details.getNextToken(doWhileSemi);
+        if (nextToken != null && (nextToken.getType() == TokenTypes.DO_WHILE
+                || nextToken.getType() == TokenTypes.LITERAL_FINALLY
+                || nextToken.getType() == TokenTypes.LITERAL_CATCH)) {
```

Comment:
Let's do a favor for the next person to come across this code: please extract these tokens to some descriptively named collection, and do something like this:
```suggestion
        final int[] myAwesomeCollection = {TokenTypes.DO_WHILE, ...};
        if (TokenUtil.isOfType(nextToken, myAwesomeCollection) {
```

I am wondering why there isn't more tokens in this condition (LITERAL_FOR, LITERAL_IF, etc.), so it would be good to point it out by the name of this collection.

Reply:
> please extract these tokens to some descriptively named collection,

@nrmancuso done I like this update but I am not sure what descriptive name could be nothing on my mind. we just checking this tokens as we have  to get parent first before getting the next token so ...? 🤔 




> I am wondering why there isn't more tokens in this condition

yea I was wondering the same but I analyzed it and it turned out that if the next token is IF or for then it is ok we don't need to update it as it will never be a second of block of related group of blocks ( no relation between If and for and its previous tokens) 
but in case that let's say the next token is a catch (the current token is try) then we have update the next token to be the one after the catch as try-catch block is treated as one block in alone or single line option consider this
```
try{ }catch{ }
```
when we want to validate that try is a block on single line we have to check that is not in the same line with the next token to its next token (it may be method braces or something else)
but consider` for( ... ) if( )` no relation between this two and they can't both be considered as block alone on single line so that's why if the next token is If or for we leave it as it is and not get its next token